app/routers.py

```diff
@@ -0,0 +1,66 @@
+from typing import List
+
+from fastapi import APIRouter, Depends, Query
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from app.crud import (
+    get_all_cities,
+    create_new_city,
+    update_city_by_id,
+    get_city_by_id,
+    delete_city_by_id,
+    get_all_temperatures,
+    get_temperatures_by_city,
+    update_all_temperatures
+)
+from app.schemas import City, CityCreate, Temperature
+from dependencies import get_db
+
+router = APIRouter()
+
+
+def get_pagination(skip: int = 0, limit: int = 10):
+    return {"skip": skip, "limit": limit}
+
+
+@router.get("/cities/", response_model=List[City])
+async def read_cities(
+        pagination: dict = Depends(get_pagination),
+        db: AsyncSession = Depends(get_db)
+):
+    skip = pagination["skip"]
+    limit = pagination["limit"]
+    return await get_all_cities(db=db, skip=skip, limit=limit)
+
+
+@router.get("/cities/{city_id}", response_model=City)
+async def read_city(city_id: int, db: AsyncSession = Depends(get_db)):
+    return await get_city_by_id(db=db, city_id=city_id)
+
+
+@router.post("/cities/", response_model=CityCreate)
+async def create_city(city: CityCreate, db: AsyncSession = Depends(get_db)):
+    return await create_new_city(db=db, city=city)
```

Comment:
You might want to handle the case when the city already exists. Currently, if a city with the same name is posted, it will be added as a new entry in your database.